File: Sorting Algorithms/insertion_sort.py
# Function definition for Insertion Sort
def insertion_sort(arr):
    n = len(arr)
    
    # Traversing through all the elements of the array
    # All the elements at index < i are present in sorted order
    for i in range(1, n):
        val = arr[i]
        
        # Move the elements of arr[0:i-1] one position ahead which are greater than val
        j = i-1
        while j>=0 and val<arr[j]:
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = val
        
def main():
    arr = [64, 34, 25, 12, 22, 11, 90]
    insertion_sort(arr)
    
    print("Sorted Array: ")
    print(*arr)
    
if __name__ == '__main__':
    main()
    
        

File: Sorting Algorithms/insertion_sort.java
class InsertionSort { 
    void sort(int arr[]) { 
        for (int i = 1; i < arr.length; i++) {
            int current = arr[i];
            int j = i - 1;
            while(j >= 0 && current < arr[j]) {
                arr[j+1] = arr[j];
                j--;
            }
            arr[j+1] = current;
        }
    } 
    public static void main(String args[]) { 
        int arr[] = {16, 53, 48, 62, 5}; 
        InsertionSort insertionSort = new InsertionSort(); 
        insertionSort.sort(arr); 
    } 
}  